src/current/v24.3/triggers.md

```diff
@@ -0,0 +1,505 @@
+| `TB_TABLE_SCHEMA` | `NAME`     | Name of the table schema tassociated with the triggerk.                                                                                                                               |
+| `TG_NARGS`        | `INT`      | Number of arguments passed to the trigger function in the [`CREATE TRIGGER`]({% link {{ page.version.version }}/create-trigger.md %}) definition.                                     |
+| `TG_ARGV`         | `STRING[]` | Arguments passed to the trigger function in the [`CREATE TRIGGER`]({% link {{ page.version.version }}/create-trigger.md %}) definition.                                               |
+
+## Examples
+
+### Create an audit log
+
+In the following example, a trigger is used to log data changes to an "audit log" table.
+
+1. Run [`cockroach demo`]({% link {{ page.version.version }}/cockroach-demo.md %}) to start a temporary, in-memory cluster with the [`movr`]({% link {{ page.version.version }}/movr.md %}) sample dataset preloaded:
+
+	{% include_cached copy-clipboard.html %}
+	~~~ shell
+	cockroach demo
+	~~~
+
+1. Create a table that stores audit records. Each record includes the table that was affected, the SQL operation that was performed on the table, the old and new table rows, and the timestamp when the change was made:
+
+	{% include_cached copy-clipboard.html %}
+	~~~ sql
+	CREATE TABLE audit_log (
+	    id UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+	    table_name TEXT NOT NULL,
+	    operation TEXT NOT NULL,
+	    old_data JSONB,
+	    new_data JSONB,
+	    changed_at TIMESTAMP DEFAULT current_timestamp
+	);
+	~~~
+
+1. Create a [trigger function]({% link {{ page.version.version }}/create-function.md %}#create-a-trigger-function) that inserts the corresponding values into the `audit_log` table:
+
+	{% include_cached copy-clipboard.html %}
+	~~~ sql
+	CREATE OR REPLACE FUNCTION audit_changes()
+	RETURNS TRIGGER AS $$
+	BEGIN
+	    INSERT INTO audit_log (table_name, operation, old_data, new_data, changed_at)
+	    VALUES (TG_TABLE_NAME, TG_OP, row_to_json(OLD), row_to_json(NEW), current_timestamp);
+	    RETURN NULL;
+	END;
+	$$ LANGUAGE PLpgSQL;
+	~~~
+
+	This function inserts the following [trigger variables](#trigger-variables):
+	- `TG_TABLE_NAME`: The table associated with the trigger. In this example, this will be `users`.
+	- `TG_OP`: The SQL operation that was performed on the table, thus activating the trigger.
+	- `OLD`: The old table row affected by `UPDATE` and `DELETE` operations.
+	- `NEW`: The new table row resulting from `INSERT` and `UPDATE` operations.
+		
+	`current_timestamp` generates a new timestamp each time the function is executed by the trigger.
+
+1. Create a trigger that executes the `audit_changes` function after an `INSERT`, `UPDATE`, or `DELETE` is issued on the `users` table:
```

Comment:
A suggestion you could take or leave: this could be a good point to demonstrate that the trigger function can be used for multiple tables, and that it will work to log mutations on a different table as well.